raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -108,6 +109,36 @@ def is_valid_refund(original_transfer, refund_sender, refund_transfer):
     )
 
 
+def is_valid_refund2(
+        original_transfer: LockedTransferUnsignedState,
+        refund_transfer: LockedTransferState):
+    """ True if the refund transfer matches the original transfer. """
+    refund_transfer_sender = refund_transfer.balance_proof.sender
+
+    # Ignore a refund from the target
+    if refund_transfer_sender == original_transfer.target:
+        return False
+
+    return (
+        original_transfer.identifier == refund_transfer.identifier and
+        original_transfer.lock.amount == refund_transfer.lock.amount and
+        original_transfer.lock.hashlock == refund_transfer.lock.hashlock and
+        original_transfer.target == refund_transfer.target and
```

Comment:
Shouldn't this be `refund_transfer.sender`?

Reply:
Nope, this is the transfer target, not the message receipient.

example:

A -> B -> C -> B ... T

C sends a refund to B,
The transfer target is T, and it must not change, this is the check above.
The transfer sender is C, the transfer receipient is B